### DISCUSSION TITLE
Difference between Classfication.MulticlassJaccardIndex (weighted = "macro") and segmentation.MeanIoU

### DISCUSSION THREAD

**Discussion by DeShinHwa** [Lightning-AI/torchmetrics#2576] [Q&A] [answered]
Hi community, may I ask what is the difference between `Classfication.MulticlassJaccardIndex(weighted="macro")` and `segmentation.MeanIoU`. Do both actually interchangeable?

**Comment by Borda** [comment, marked as answer]
The main difference is their typical use case and input format. `Classification.MulticlassJaccardIndex` (weighted=“macro”) computes class-wise Jaccard index averaged over classes, mainly for classification tasks with discrete labels. `segmentation.MeanIoU` is designed for pixel-wise segmentation masks and averages IoU across classes.
Both measure intersection-over-union but differ in expected inputs and context, so they aren’t fully interchangeable. Use the one that matches your task’s data format and prediction type.
